include/ginkgo/core/base/polymorphic_object.hpp

```diff
@@ -228,6 +251,21 @@ class PolymorphicObject : public log::EnableLogging<PolymorphicObject> {
         return copied;
     }
 
+    template <typename Derived>
+    std::enable_if_t<
+        std::is_base_of<PolymorphicObject, std::decay_t<Derived>>::value,
+        PolymorphicObject>*
+    move_from(const std::unique_ptr<Derived>& other)
```

Comment:
Isn't it possible to merge all the `move_from` functions into just `enable_if<...> move_from(pointer_param<Derived> other)`? Essentially, they are all doing the same. Only if a user has overridden the default implementation of `copy_from_impl(unique_ptr)` there might be a difference.

Reply:
Also, apologies for implementing`move_from(unique_ptr)` via the copy function instead of the move function. In hindsight, that was the wrong choice.